### PR TITLE
light floors emit colored lights

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: lightfloor makes colored lights and are more visible
fix: you can turn them off/on by using your screwdriver
fix: you can emp em
fix: breaking them actually works
fix: using a broken bulb wont fix em
sprites: flicker states for each colour
/:cl:

the "black" light doesnt make sense so i made it able to turn off and on
the "s" state was literally the same as color cycling just with some colors moved so i removed it

/obj/item/stack/tile/light/Initialize(mapload, new_amount, merge = TRUE) <--- this code is super dumb it makes light tiles BREAK WHILE STACK MERGING N STUFF so i removed it
inb4 .dmi conflicts

### CODE REVIEW

File: code/game/turfs/open/floor/light_floor.dm [hunk 1]
```diff
@@ -2,34 +2,42 @@
...
+	icon_state = "light_on-1"
 	floor_tile = /obj/item/stack/tile/light
 	broken_states = list("light_broken")
+	///var to see if its on or off
 	var/on = TRUE
-	var/state = 0//0 = fine, 1 = flickering, 2 = breaking, 3 = broken
-	var/static/list/coloredlights = list("r", "o", "y", "g", "b", "i", "v", "w", "s", "z")
-	var/currentcolor = "b"
+	///0 = fine, 1 = flickering, 2 = breaking, 3 = broken
+	var/state = 0
```
Comment (Qustinnus) [inline code/game/turfs/open/floor/light_floor.dm:11]: this should use defines

Comment (trollbreeder) [reply code/game/turfs/open/floor/light_floor.dm:11]:
```
	#define LIGHTFLOOR_FINE 0
	#define LIGHTFLOOR_FLICKER 1
	#define LIGHTFLOOR_BREAKING 2
	#define LIGHTFLOOR_BROKEN 3

	var/state = LIGHTFLOOR_FINE
```

Comment (Qustinnus) [reply code/game/turfs/open/floor/light_floor.dm:11]: tlalt knows how to use defines

**Comment by RaveRadbury** [comment]
Tlal, I just did the PR that added that stuff. The offset is so you can have alternating floors. The black sprite is so you can easily set up color systems. It's a QoL update.

I made #50966 recently

image

I request that you do not include these changes in your PR.

**Comment by Tlaltecuhtli** [comment]
u can have your alt color but having a fake turned off black tile is dumb i dont see how its a qol for someone to expect the black tile to not emit light but it does
 having togglable by hand click is more qol especially if some traitor wants some darkness quickly without having to fiddle with a vibrator wand

**Comment by Fikou** [comment]
black light moment

**Comment by RaveRadbury** [comment]
> having togglable by hand click 

The whole problem is that people walk onto these floors and accidentally click them. That's why I made it multitool only. Also, what traitor is in need of turning of dance floor light tiles??????

I made the previous PR with the intention of making light tiles only alterable with a multitool. The reasoning for that is obvious.

**Comment by Tlaltecuhtli** [comment]
if someone accidentally click he can just click back, nightmares dont have multitools

**Comment by nemvar** [comment]
put it on alt-click or something. just plain attack_hand sounds silly.

**Comment by RaveRadbury** [comment]
> if someone accidentally click he can just click back, nightmares dont have multitools

Nightmares also have a blade that kills any light device, they don't need to flick a switch on a dance floor.

> put it on alt-click or something. just plain attack_hand sounds silly.

This almost works, but alt-clicking still has a bunch of other usages.

**Review by kriskog — DISMISSED** [review]
.

**Review by Rohesie — CHANGES_REQUESTED** [review]
Do the changes requested by RaveRadbury please.

**Comment by Rohesie** [comment]
@RaveRadbury ?

**Comment by RaveRadbury** [comment]
Putting the on/off switch onto a screwdriver is fine. Tlal seems to be committed to the idea of having the black color setting be a separate and literal switch. Given the improvements that he has made to the light tiles here I am alright with this change, as ultimately it's a realism vs ease argument.

I don't quite understand why all the colors have unique names when they are quite literally ROYGBIV but it's not really an issue.

Reviewing the icon changes, light_on_flicker-9 and light_on_flicker-10 have a stray yellow pixel in the upper right of several frames. This should be addressed.

Other than the stray pixels this all looks good. Sorry I just got to this now.

**Comment by Rohesie** [comment]
Could you fix the stray pixels then? We can merge after that.

**Comment by Tlaltecuhtli** [comment]
i checked the icons and i see no yellow pixel in light_on_flicker-9 and light_on_flicker-10

**Comment by RaveRadbury** [comment]
image

It's there, on multiple frames.

**Comment by Tlaltecuhtli** [comment]
there are no white tile sprites on -9 and -10 (multicolors), even -8 (the white tile) doesnt have it are you sure you havent missclicked a pixel on your local?

**Comment by RaveRadbury** [comment]
I downloaded this from the check tab, is there a reason why the check tab would throw this kind of error?

**Comment by RaveRadbury** [comment]
@Tlaltecuhtli I checked out your PR on local and I didn't see the pixels. It seems that it is an issue with the icon-diff bot. Apologies for hanging up your PR on this.

@Rohesie It's good to go

**Pull request merged**